### PR TITLE
Capture permission error on windows

### PR DESCRIPTION
**Pull request by oeway**
This PR catches error from `os.listdir` (tested on windows), and added instructions for installing a custom version of jupyter-server-proxy with windows support.

### CODE REVIEW

**Comment by oeway** [comment]
For the `self.__files(path)` called in `open` command, there is a folder passed the check of `self.__is_allowed(path, "read")`, but yield a permission error with `os.listdir(path)` on Windows. Any idea? @MartinHjelmare

**Comment by MartinHjelmare** [comment]
Not sure. There's this note in the std lib docs:
> Note I/O operations may fail even when access() indicates that they would succeed, particularly for operations on network filesystems which may have permissions semantics beyond the usual POSIX permission-bit model.

https://docs.python.org/3/library/os.html#os.access

File: jupyter_elfinder/elfinder.py [hunk 1]
```diff
@@ -1200,15 +1206,28 @@ def __ls(self) -> None:
...
+                self._response[R_ERROR] = "Directory (link) not found"
+                return
+
+        if not self.__is_allowed(path, "read"):
+            self._response[R_ERROR] = "Access denied"
+            return
+
+        try:
+            items = {}
+            for fname in os.listdir(path):
```
Comment (MartinHjelmare) [inline jupyter_elfinder/elfinder.py]: It's easier to read if we limit the scope of the try... except as much as possible by only wrapping the line(s) that can raise. We also save on indentation.

```py
items = {}
try:
    file_names = os.listdir(path)
except PermissionError:
    self._response[R_ERROR] = "Access denied"
    return
for fname in file_names:
    ...
```

**Comment by oeway** [comment]
Alright, to conclude a bit, there will be two more checks which we need to systematically do for all the requests:
1. check if the file name from the client is safe
2. capture error even if we have checked with `os.access`.

**Comment by MartinHjelmare** [comment]
Maybe we can eventually drop the access check altogether and just catch `PermissionError`?

Particularly on file access that is recommended:
https://docs.python.org/3/glossary.html#term-eafp

**Pull request merged**